### PR TITLE
CL: Initialize tick minor gas optimization & cleanup

### PR DESCRIPTION
**Pull request by ValarDragon**
This PR is simplifying the `GetTickInfo` code, by moving the initialization logic to its own function. Furthermore, we only get the pool once, removing repeated gets in every function this would otherwise call. (Saving a net of 2 GetPool operations). This has an added benefit of removing messiness from the test code, where it attempted to generate test cases with invalid pool ID's.

The remainder of this PR is cleaning up tests.
The "TestGetInitialSpreadRewardGrowthOppositeDirectionOfLastTraversalForTick" was extremely messy, so just re-wrote the edge case that idt even makes sense to test into its own function, and made the main test use standard helpers.

### CODE REVIEW

File: x/concentrated-liquidity/fuzz_test.go [hunk 1]
```diff
@@ -40,7 +40,7 @@ type positionAndLiquidity struct {
 }
 
 func TestFuzz_Many(t *testing.T) {
-	fuzz(t, defaultNumSwaps, defaultNumPositions, 100)
+	fuzz(t, defaultNumSwaps, defaultNumPositions, 10)
```
Comment (ValarDragon) [inline x/concentrated-liquidity/fuzz_test.go:43]: Driveby change, now that were through CL, running 100 iterations on every save no longer make sense. (It also makes test on save expensive!)

File: x/concentrated-liquidity/tick.go [hunk 2]
```diff
@@ -125,38 +125,41 @@ func (k Keeper) GetTickInfo(ctx sdk.Context, poolId uint64, tickIndex int64) (ti
...
-		// Initialize uptime trackers for the new tick to the appropriate starting values.
-		valuesToAdd, err := k.getInitialUptimeGrowthOppositeDirectionOfLastTraversalForTick(ctx, poolId, tickIndex)
-		if err != nil {
-			return tickStruct, err
-		}
+func (k Keeper) makeInitialTickInfo(ctx sdk.Context, poolId uint64, tickIndex int64) (tickStruct model.TickInfo, err error) {
+	pool, err := k.getPoolById(ctx, poolId)
+	if err != nil {
+		return tickStruct, err
+	}
```
Comment (ValarDragon) [inline x/concentrated-liquidity/tick.go:138]: The core change, and using pool as the arg for the next 3 functions

**Review by AlpinYukseloglu — APPROVED** [review]
LGTM, nice work!
